features/0434-outofband/README.md

```diff
@@ -0,0 +1,417 @@
+- `service` - an item that is the equivalent of the service block of a DIDDoc that the receiver is to use in responding to the message. Additional details below.
+
+##### The `service` Item
+
+As mentioned in the description, the `service` item array is intended to be analogous to the `service` block of a DIDDoc. There are two forms of entries in the `service` item array:
+
+- a public DID that is resolved to retrieve it's DIDDoc service block, and
+- an inline service block.
+
+The following is an example of a two entry array, one of each form:
+
+```jsonc
+{
+  "@type": "https://didcomm.org/outofband/VER/invitation",
+  "@id": "<id used for context as pthid>",
+  "label": "Faber College"
+  "protocols": ["https://didcomm.org/didexchange/1.0"]
+  "service": [
+      {
+        "id": "#inline"
+        "type": "did-communication",
+        "recipientKeys": ["did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH"],
+        "routingKeys": [],
+        "serviceEndpoint": "https://example.com:5000"
+      },
+      "did:sov:LjgpST2rjsoxYegQDRm7EL"
+  ]
+}
+```
+
+The processing rules for the service block are:
+
+- Use only entries where the `type` is equal to `did-communication`.
+  - Entries without a `type` are assumed to be "did-communication".
+- The sender **MUST** put the array into their order of preference, and, as such, the receiver **SHOULD** prefer the entries in order.
+
+The attributes in the inline form parallel the attributes of a DID Document for increased meaning. The `recipientKeys` and `routingKeys` within the inline block decorator must be [`did:key` references](https://digitalbazaar.github.io/did-method-key/).
+
+As defined in the [DIDComm Cross Domain Messaging RFC](https://github.com/hyperledger/aries-rfcs/tree/master/concepts/0094-cross-domain-messaging), if `routingKeys` is present and non-empty, additional forwarding wrapping are necessary in the response message.
+
+When considering routing and options for out of band messages, keep in mind that the more detail in the message, the longer the URL will be and (if used) the more dense (and harder to scan) the QR code will be.
+
+###### Service Endpoint
+
+The service endpoint used to transmit the response is either present in the out of band message or available in the DID Document of a presented DID. If the endpoint is itself a DID, the `serviceEndpoint` in the DIDDoc of the resolved DID **MUST** be a URI, and the `recipientKeys` must contain a single key. That key is appended to the end of the list of `routingKeys` for processing. For more information about message forwarding and routing, see [RFC 0094 Cross Domain Messaging](https://github.com/hyperledger/aries-rfcs/tree/master/concepts/0094-cross-domain-messaging).
+
+### Adoption Messages
+
+The `problem_report` message **MAY** be adopted by the out of band protocol if the agent wants to respond with problem reports to invalid messages, such as attempting to reuse a single-use invitation.
```

Comment:
I don't see how problem-report messages could be used if you don't yet have a connection. How would you send them? Over QR codes?

Reply:
I tend to agree, but in the case mentioned (all the requirements met), they have sent you their DIDDoc for the relationship, and in that case you COULD respond with a DIDComm message.  Again - it's a pretty limited case.

Reply:
Okay. Thanks for the clarification. Point withdrawn.